.tools/validate.go

```diff
@@ -0,0 +1,260 @@
+	Author               PersonAction `json:"author,omitempty"`
+	Commiter             PersonAction `json:"commiter,omitempty"`
+}
+
+// PersonAction is a time and identity of an action on a git commit
+type PersonAction struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	Date  string `json:"date"` // this could maybe be an actual time.Time
+}
+
+var (
+	prettyLogSubject       = `--pretty=format:%s`
+	prettyLogBody          = `--pretty=format:%b`
+	prettyLogCommit        = `--pretty=format:%H`
+	prettyLogAuthorName    = `--pretty=format:%aN`
+	prettyLogAuthorEmail   = `--pretty=format:%aE`
+	prettyLogCommiterName  = `--pretty=format:%cN`
+	prettyLogCommiterEmail = `--pretty=format:%cE`
+	prettyLogFormat        = `--pretty=format:{"commit": "%H", "abbreviated_commit": "%h", "tree": "%T", "abbreviated_tree": "%t", "parent": "%P", "abbreviated_parent": "%p", "refs": "%D", "encoding": "%e", "sanitized_subject_line": "%f", "commit_notes": "%N", "verification_flag": "%G?", "signer": "%GS", "signer_key": "%GK", "author": { "date": "%aD" }, "commiter": { "date": "%cD" }}`
+)
+
+// GitLogCommit assembles the full information on a commit from its commit hash
+func GitLogCommit(commit string) (*CommitEntry, error) {
+	buf := bytes.NewBuffer([]byte{})
+	cmd := exec.Command("git", "log", "-1", prettyLogFormat, commit)
+	cmd.Stdout = buf
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		log.Println(strings.Join(cmd.Args, " "))
+		return nil, err
+	}
+	c := CommitEntry{}
+	output := buf.Bytes()
+	if err := json.Unmarshal(output, &c); err != nil {
+		fmt.Println(string(output))
+		return nil, err
+	}
+
+	output, err := exec.Command("git", "log", "-1", prettyLogSubject, commit).Output()
+	if err != nil {
+		return nil, err
+	}
+	c.Subject = strings.TrimSpace(string(output))
+
+	output, err = exec.Command("git", "log", "-1", prettyLogBody, commit).Output()
+	if err != nil {
+		return nil, err
+	}
+	c.Body = strings.TrimSpace(string(output))
+
+	output, err = exec.Command("git", "log", "-1", prettyLogAuthorName, commit).Output()
+	if err != nil {
+		return nil, err
+	}
+	c.Author.Name = strings.TrimSpace(string(output))
+
+	output, err = exec.Command("git", "log", "-1", prettyLogAuthorEmail, commit).Output()
+	if err != nil {
+		return nil, err
+	}
+	c.Author.Email = strings.TrimSpace(string(output))
+
+	output, err = exec.Command("git", "log", "-1", prettyLogCommiterName, commit).Output()
+	if err != nil {
+		return nil, err
+	}
+	c.Commiter.Name = strings.TrimSpace(string(output))
+
+	output, err = exec.Command("git", "log", "-1", prettyLogCommiterEmail, commit).Output()
+	if err != nil {
+		return nil, err
+	}
+	c.Commiter.Email = strings.TrimSpace(string(output))
+
+	return &c, nil
+}
+
+// GitCommits returns a set of commits. If commitrange is a git still range 12345...54321, the it will be isolated set of commits. If commitrange is a single commit, the set will be beginning commit up to the hash provided.
```

Comment:
“the it will” → “then it will”.

And “the set will be beginning commit up to the hash provided” → “all ancestor commits up through the hash provided” (because there may be more than one parent-less commit in a given commit's ancestry, e.g. if you'd merged an orphan branch).